### DISCUSSION TITLE
About the Dynamic Heterogeneous Graph

### DISCUSSION THREAD

**Discussion by BlakeQG** [pyg-team/pytorch_geometric#8693] [Q&A] [answered]
We are planning to explore PyG’s HeteroConv/HGTConv for our use case (similar to the example given in the original HGT paper, a dynamic heterogeneous graph).  By analogy, we want to find PhDs authors at 2023, e.g., `MichaelDavis@2023` similar to `JureLeskovec@2012`. 

I am a bit confused how the [ HeteroConv/HGTConv example](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/heterogeneous.html) in the PyG maps to the example in the HGT paper in terms of “**dynamic**”. In the HGT paper it claims it can handle dynamic graph, e.g., associating an event node with different timestamps such as “WWW@2015”, “WWW@2016”, etc. Does it mean creating individual event node such as “WWW@2015”, “WWW@2016”, ... in the graph (as I don’t see such thing in the PyG example) ? 
     Or there is only one WWW node? So in order to compute the embedding of WWW@2015, I need to take a [snapshot](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.data.HeteroData.html#torch_geometric.data.HeteroData.up_to) of the HeteroData at 2015 to compute its embedding, then snapshot at 2016 for WWW@2016, then snapshot at 2017 for WWW@2017,....?

**Comment by rusty1s** [comment, marked as answer]
Temporal information is added via relative positional encoding to the node representations, as discussed in Section 3.5. One can achieve this via the usage of `torch_geometric.nn.PositionalEncoding`. However, it is not (yet) integrated into our `HGTConv` implementation.

**Reply by KilianZimmerer** [reply]
Since we need this feature in our current project, I could contribute an implementation of relative temporal encoding for HGTConv. Would this be beneficial to the project?

**Reply by KilianZimmerer** [reply]
PR that enables temporal encoding for HGTConv: https://github.com/pyg-team/pytorch_geometric/pull/10469
